### ISSUE TITLE
"Not enough memory resources are available to complete this operation"

### ISSUE THREAD

**Issue by doremyy** [Legacy-LuaSTG-Engine/LuaSTG-Sub#22]
i've been facing this strange error during my experience in using LSTGSub. It did not exactly happen before, but it's a prevalent issue ever since, and it prevents me from loading my projects and testing what I have created. even just lauching the vanilla LuaSTGSub.exe to see the demo mode, the same error is given to me. my laptop specs are included in the engine.log file attached.

[engine.log](https://github.com/Legacy-LuaSTG-Engine/LuaSTG-Sub/files/13621694/engine.log)

**Comment by Demonese** [comment]
This looks strange, it may be an AMD graphics card driver issue. You can try configuring LuaSTGSub.exe to use NVIDIA dedicate graphics card in the Windows settings (System - Display).

**Comment by doremyy** [comment]
I configured it to use the NVIDIA GPU, and it seems to at the very least fix the engine being unable to initialize. The issue is still prevalent in some other cases, indicated by this log which was produced after the engine froze when trying to test a spellcard.

[engine.log](https://github.com/Legacy-LuaSTG-Engine/LuaSTG-Sub/files/13625887/engine.log)

Is it an issue that AMD has to fix on their end? Or is there some sort of compatibility issue that lua has?

**Comment by Demonese** [comment]
> The issue is still prevalent in some other cases

Yes, the problem now is not enough memory. The graphics card works normally.

Currently, the executable file of LuaSTG is 32-bit, so the available memory is limited to 2GB, even if the host has a large amount of available memory. However, other people in our community rarely see the problem of not enough memory.

If create a new blank spellcard project, can it run normally?

**Comment by Demonese** [comment]
I compile a new executable. You might try this, although it may not solve your problem
[(preview) LuaSTG-Sub-v0.20.16.zip](https://github.com/Legacy-LuaSTG-Engine/LuaSTG-Sub/files/13626308/preview.LuaSTG-Sub-v0.20.16.zip)

**Comment by doremyy** [comment]
it does not solve my issue, even with the new executable. other spellcards that i have created work just fine, so i wonder if it's a particular issue with what i've created...
